docs/tasks/administer-cluster/dns-custom-nameservers.md

```diff
@@ -0,0 +1,150 @@
+nameservers.
+{% endcapture %}
+
+{% capture prerequisites %}
+* {% include task-tutorial-prereqs.md %}
+* The cluster must be configured to use the `kube-dns` addon.
+{% endcapture %}
+
+{% capture steps %}
+
+## Name resolution in Kubernetes
+
+The diagram below shows the flow of DNS queries specified in the configuration
+above. With the dnsPolicy set to “ClusterFirst” a DNS query is first sent to
+the DNS caching layer in kube-dns. From here, the suffix of the request is
+examined and then forwarded to the appropriate DNS.  In this case, names with
+the cluster suffix (e.g.; “.cluster.local”) are sent to kube-dns. Names with
+the stub domain suffix (e.g.; “.acme.local”) will be sent to the configured
+custom resolver. Finally, requests that do not match any of those suffixes will
+be forwarded to the upstream DNS.
+
+![DNS lookup flow](/docs/tasks/administer-cluster/dns-custom-nameservers/dns.png)
+
+## Configuring stub-domain and upstream DNS servers
+
+Cluster administrators can specify custom stub domains and upstream nameservers
+by providing a ConfigMap for kube-dns (`kube-system:kube-dns`).
+
+For example, the configuration below inserts a single stub domain and two
+upstream nameservers.  As specified, DNS requests with the “.acme.local” suffix
+will be forwarded to a DNS listening at 1.2.3.4. Additionally, Google Public DNS
+will serve upstream queries. See the [ConfigMap options](#configmap-options) for
+details about the configuration option format.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-dns
+  namespace: kube-system
+data:
+  stubDomains: |
+    {“acme.local”: [“1.2.3.4”]}
+  upstreamNameservers: |
+    [“8.8.8.8”, “8.8.4.4”]
+```
+
+The diagram below shows the flow of DNS queries specified in the configuration
+above. With the dnsPolicy set to “ClusterFirst”, a DNS query is first sent to
+the DNS caching layer in kube-dns. From here, the suffix of the request is
+examined and then forwarded to the appropriate DNS.  In this case, names with
+the cluster suffix (e.g. “.cluster.local”) are sent to kube-dns. Names with the
+stub domain suffix (e.g. “.acme.local”) will be sent to the configured custom
+resolver. Finally, requests that do not match any of those suffixes will be
+forwarded to the upstream DNS.
+
+Below is a table of example domain names and the destination of the queries for
+those domain names:
+
+| Domain name | Server answering the query |
+| ----------- | -------------------------- |
+| kubernetes.default.svc.cluster.local| kube-dns |
+| foo.acme.local| custom DNS (1.2.3.4) |
+| widget.com    | upstream DNS (one of 8.8.8.8, 8.8.4.4) |
+
+{% endcapture %}
+
+{% capture discussion %}
+
+## Understanding custom DNS upstream servers and stub domains
+
+### Pod DNS policies
+
+Kubernetes currently supports two DNS policies specified on a per-pod basis
+using the dnsPolicy flag: “Default” and “ClusterFirst”. If dnsPolicy is not
+explicitly specified, then “ClusterFirst” is used:
```

Comment:
An unrelated comment: Making "ClusterFirst" the default value while having another option called "Default" looks really confusion :)